app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsAdvancedScreen.kt

```diff
@@ -764,6 +800,67 @@ object SettingsAdvancedScreen : SearchableSettings {
         )
     }
 
+    private suspend fun testFlareSolverrAndUpdateUserAgent(
+        flareSolverrUrlPref: BasePreference<String>,
+        userAgentPref: BasePreference<String>,
+        context: android.content.Context
+    ) {
+        try {
+            withContext(Dispatchers.IO) {
+                val client = OkHttpClient()
+                val flareSolverUrl = flareSolverrUrlPref.get().trim()
+                val mediaType = "application/json; charset=utf-8".toMediaType()
+                val data = JSONObject()
+                    .put("cmd", "request.get")
+                    .put("url", "https://www.google.com/")
+                    .put("maxTimeout", 60000)
+                    .put("returnOnlyCookies", true)
+                    .toString()
+                val body = data.toRequestBody(mediaType)
+                val request = Request.Builder()
+                    .url(flareSolverUrl)
+                    .post(body)
+                    .header("Content-Type", "application/json")
+                    .build()
+
+                Log.d("FlareSolverrRequest", "Sending request to FlareSolverr: $flareSolverUrl with payload: $data")
+
+                val response = client.newCall(request).execute()
```

Comment:
Use .awaitSuccess() instead of execute(), this way you don't have to manually check the success as well